src/Reports/Reports.tsx

```diff
@@ -1,11 +1,89 @@
-import React, { Component } from "react";
+import React from "react";
+import { BarChart, Bar, CartesianGrid, XAxis, YAxis, Label, Cell, Line, ComposedChart } from 'recharts';
+import { JournalEntry } from "../journal";
+import { JournalContext, useJournal } from "../JournalContext";
+import { scaleLinear } from 'd3-scale';
 
-type ReportsProps = {
-  //
+//color gradient for bar graph
+const colorScale = scaleLinear<string>()
+  .domain([1, 3, 5, 10, 18]) // 1 is red, 3 is orange, 5 is yellow, 10 is green, all the way up is more green
+  .range(["#cf2900","#e68620", "#fef200", "#34ba36", "#12a114"]);
+
+//graphs
+export const LineChartComponent: React.FC<{
+  reportGraph: JournalEntry[];
+  dataKey: string;
+  yAxisLabel: string;
+
+}> = ({ reportGraph, dataKey, yAxisLabel}) => {
```

Comment:
I would chage this starting at line 18 to reverse the order in which data points are displayed:

 }> = ({ reportGraph, dataKey, yAxisLabel}) => {
const graphData = [...reportGraph].reverse(); // use graphData as reversed array so data appears chronologically

  return (
    <ComposedChart
      width={500}
      height={300}
      barGap={10}
      barSize={80} 
      data={graphData}
      margin={{
        top: 5,
        right: 30,
        left: 20,
        bottom: 5,
      }}

Reply:
if this is something we always wanted the component to do then I would agree but for this instance we are rendering the same dataset multiple times which would mean we are reversing the same list x times when we can just reverse it once and pass that to all the components.